packages/components/src/internal/components/domainproperties/BaseDomainDesigner.tsx

```diff
@@ -137,7 +140,10 @@ export const BaseDomainDesigner: FC<BaseDomainDesignerProps> = memo(props => {
         onCancel,
         hasValidProperties,
         saveBtnText = 'Save',
+        showUserComment
     } = props;
+    const [userComment, setUserComment] = useState<string>(undefined);
+    const requiresUserComment = showUserComment ? useDataChangeCommentsRequired().requiresUserComment : false;
```

Comment:
We should really try to avoid conditional calls to hooks. Additionally, `showUserComment` is always true when rendered below so I think this should be:

```tsx
const { requiresUserComment } = useDataChangeCommentsRequired();

// ...

{showUserComment && (
    <CommentTextArea
        actionName="Update"
        containerClassName="inline-comment"
        onChange={setUserComment}
        requiresUserComment={requiresUserComment}
        inline
    />
)}
```

Reply:
I was trying to avoid `useDataChangeCommentsRequired()` hook usage when in LKS. If you have better ideas how to handle this I'm all for it. We have a backlog item to enforce api level required change comment, if helps.

Reply:
Comment added